### PR TITLE
Make max_subcompactions dynamically changeable

### LINKED ISSUE #13145: Make max_subcompactions dynamically changeable

### PR DESCRIPTION
**Pull request by ethercflow**
### What is changed and how it works?

Issue Number: Close #13145

What's Changed:

```commit-message
Make max_subcompactions dynamically changeable
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

- Performance regression
    - None

### Release note 

```release-note
Make max_subcompactions dynamically changeable
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1615,6 +1649,14 @@ impl ConfigManager for DBConfigManger {
             self.set_max_background_flushes(max_background_flushes)?;
         }
 
+        if let Some(background_subcompactions_config) = change
+            .drain_filter(|(name, _)| name == "max_sub_compactions")
+            .next()
+        {
+            let max_subcompactions = background_subcompactions_config.1.into();
+            self.set_max_subcompactions(max_subcompactions)?;
+        }
```
Comment (tabokie) [inline src/config.rs:1658]: Add a test?

**Review by tabokie — APPROVED** [review]

File: src/config.rs (hunk 1 above)
Comment (ethercflow) [reply src/config.rs:1658]: I think no need? As https://github.com/tikv/rocksdb/blob/6.4.tikv/db/db_compaction_test.cc#L4943 already has tests.
Just passthrough here.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**